test/mpfr.jl

```diff
@@ -1,898 +1,1016 @@
-f = BigFloat(10)^Int32(1000)
-@test i1 != i2
-@test i1 != f
-@test i2 != f
+@testset "hypot" begin
+    @test hypot(BigFloat(3), BigFloat(4)) == 5
+end
 
-@test f > i1
-@test f > i2
+@testset "atan2" begin
+    setprecision(53) do
+        @test atan2(12,2) == atan2(BigFloat(12), BigFloat(2))
+    end
+end
 
-i3 = trunc(Integer,f)
-@test i3 == f
-@test i3+1 > f
-@test i3+1 >= f
+@testset "ldexp" begin
+    setprecision(53) do
+        @test ldexp(BigFloat(24.5), 72) == ldexp(24.5, 72)
+        @test ldexp(BigFloat(24.5), Int16(72)) == ldexp(24.5, 72)
+        @test ldexp(BigFloat(24.5), -72) == ldexp(24.5, -72)
+        @test ldexp(BigFloat(24.5), Int16(-72)) == ldexp(24.5, -72)
+        @test ldexp(BigFloat(24.5), UInt(72)) == ldexp(24.5, 72)
+        @test ldexp(BigFloat(24.5), 0x48) == ldexp(24.5, 72)
+    end
+end
 
-let err(z, x) = abs(z - x) / abs(x)
-    @test 1e-60 > err(eta(parse(BigFloat,"1.005")), parse(BigFloat,"0.693945708117842473436705502427198307157819636785324430166786"))
-    @test 1e-60 > err(exp(eta(big(1.0))), 2.0)
+@testset "ceil / floor / trunc" begin
+    x = parse(BigFloat,"28273.7312487489135135135")
+    y = BigInt(28273)
+    z = BigInt(28274)
+    a = parse(BigFloat,"123456789012345678901234567890.2414")
+    b = parse(BigInt,"123456789012345678901234567890")
+    c = parse(BigInt,"123456789012345678901234567891")
+    @test ceil(x) == z
+    @test typeof(ceil(x)) == BigFloat
+    @test floor(x) == y
+    @test typeof(floor(x)) == BigFloat
+    @test trunc(x) == y
+    @test typeof(trunc(x)) == BigFloat
+
+    @test ceil(Integer,x) == z
+    @test typeof(ceil(Integer,x)) == BigInt
+    @test floor(Integer,x) == y
+    @test typeof(floor(Integer,x)) == BigInt
+    @test trunc(Integer,x) == y
+    @test typeof(trunc(Integer,x)) == BigInt
+
+    @test ceil(Int64, x) == Int64(z)
+    @test typeof(ceil(Int64, x)) == Int64
+    @test floor(Int64, x) == Int64(y)
+    @test typeof(floor(Int64, x)) == Int64
+    @test trunc(Int64, x) == Int64(y)
+    @test typeof(trunc(Int64, x)) == Int64
+
+    @test ceil(Int32, x) == Int32(z)
+    @test typeof(ceil(Int32, x)) == Int32
+    @test floor(Int32, x) == Int32(y)
+    @test typeof(floor(Int32, x)) == Int32
+    @test trunc(Int32, x) == Int32(y)
+    @test typeof(trunc(Int32, x)) == Int32
+
+    @test ceil(Int16, x) == Int16(z)
+    @test typeof(ceil(Int16, x)) == Int16
+    @test floor(Int16, x) == Int16(y)
+    @test typeof(floor(Int16, x)) == Int16
+    @test trunc(Int16, x) == Int16(y)
+    @test typeof(trunc(Int16, x)) == Int16
+
+    #@test ceil(Int8, x) == Int8(z)
+    #@test typeof(ceil(Int8, x)) == Int8
+    #@test floor(Int8, x) == Int8(y)
+    #@test typeof(floor(Int8, x)) == Int8
+    #@test trunc(Int8, x) == Int8(y)
+    #@test typeof(trunc(Int8, x)) == Int8
```

Comment:
It looks like it might have been generated by some kind of automatic code generation. In fact, quite a lot of this could be simplified using a loop over types, no?